### PR TITLE
More Miner tests

### LINKED ISSUE #479: More miner_test coverage
The following testing suggestions are taken from TODOs in builtin/miner/miner_test.go

`TestControlAddresses`
* changing worker (with delay)
* changing peer id

`TestCommitments`
* test insufficient funds when the precommit deposit is set above zero
* too early to prove sector
* seal rand epoch too old
* commitment expires before proof
* invalid seal proof (more thought needs to go into how the test should work)

### PR DESCRIPTION
**Pull request by ZenGround0**
Closes #479 

- Simple ChangePeerID happy path test
- A few new sector commitment edge case tests
- Aggregate some test ideas from todo and new test ideas into linked issue

Worker changes already handled elsewhere

### CODE REVIEW

File: actors/builtin/miner/miner_test.go [hunk 1]
```diff
@@ -1693,6 +1710,26 @@ func TestWithdrawBalance(t *testing.T) {
 	})
 }
 
+func TestChangePeerID(t *testing.T) {
+	periodOffset := abi.ChainEpoch(100)
+
+	setupFunc := func() (*mock.Runtime, *actorHarness) {
+		actor := newHarness(t, periodOffset)
```
Comment (anorth) [inline actors/builtin/miner/miner_test.go:1717]: I don't like these setup funcs generally (because they couple different cases and are easy to abuse). But another reason is that this captures the wrong `t`. It must be a parameter to `setupFunc`.

File: actors/builtin/miner/miner_test.go [hunk 2]
```diff
@@ -141,18 +141,19 @@ func TestControlAddresses(t *testing.T) {
...
 
 // Test for sector precommitment and proving.
 func TestCommitments(t *testing.T) {
 	periodOffset := abi.ChainEpoch(100)
 
 	// TODO more tests
+	// https://github.com/filecoin-project/specs-actors/issues/936
 	// - Concurrent attempts to upgrade the same CC sector (one should succeed)
-	// - Insufficient funds for pre-commit, for prove-commit
+	// - Insufficient funds for prove-commit (charged in power actor -- scenario testing candidate)
```
Comment (acruikshank) [inline actors/builtin/miner/miner_test.go:153]: Since these are now covered in an issue (Thanks!), do they still need to be documented as comments? My concern is that this becomes a redundant and eventually inconsistent work queue.

Comment (ZenGround0) [reply actors/builtin/miner/miner_test.go:153]: Yeah great point, I will delete
